Version 1.0.0

yakupkavak opened the pull request.

yakupkavak committed:
feat : design game screen and bottomnavigation

yakupkavak committed:
feat : add gameId

yakupkavak committed:
refactor: refactor general

yakupkavak committed:
ignore: ---

yakupkavak committed:
feat: complete service and repository

yakupkavak committed:
ignore: ---

yakupkavak committed:
feat: get data

yakupkavak committed:
feat: upgrade family question

yakupkavak committed:
feat: ask question

yakupkavak committed:
feat: ask akatsuki family question

yakupkavak committed:
ignore: ---

yakupkavak committed:
feat: ask clan question

yakupkavak committed:
feat: ask team question

yakupkavak committed:
feat: ask tail question

scotoglu commented on app/src/main/java/com/example/narutoquiz/data/base/BaseRepository.kt:
```diff
@@ -13,9 +13,13 @@ abstract class BaseRepository {
                 return Resource.error(null)
             }
         } catch (e: Exception) {
+            println(e.message)
```
exception olursa, detaylıca görebilmek için `printStackTrace` fonksiyonu da kullabilir.

scotoglu commented on app/src/main/AndroidManifest.xml:
```diff
@@ -2,6 +2,8 @@
+    <uses-permission android:name="android.permission.INTERNET"/>
+
     <application
         android:name=".ui.main.HiltActivityClass"
```
NarutoQuizApp olarak isimlendirebiliriz. `HiltActivityClass` ile sadece Hilt için eklemişiz gibi duruyor. App genelini kapsayan init işlemleri için burası kullanılabiliyor. Örneğin Firebase init işlemleri.

scotoglu commented on app/src/main/java/com/example/narutoquiz/data/model/Age.kt:
```diff
@@ -0,0 +1,20 @@
+    @JsonProperty("Chunin Promotion")
+    val chuninPromotion: String?,
+    @JsonProperty("Part I")
+    val partI: String?,
```
partOne, partTwo olarak isimlendirebiliriz.

scotoglu commented on app/src/main/java/com/example/narutoquiz/data/model/Akatsuki.kt:
```diff
@@ -0,0 +1,14 @@
+    @JsonProperty("akatsuki")
+    val akatsuki: List<Character>,
+    @JsonProperty("currentPage")
+    val currentPage: Int,
```
Response ilgili alan null gelebilir. Önlem almakta fayda var, nullable yapabilirsin istersen.

yakupkavak replied:
o zaman her data class'ın her objesini null yapmak mı gerekir

scotoglu commented on app/src/main/java/com/example/narutoquiz/data/model/Character.kt:
```diff
@@ -0,0 +1,30 @@
+    @JsonProperty("tools")
+    val tools: List<String>?,
+    @JsonProperty("uniqueTraits")
+    val uniqueTraits: List<String>?,
```
👍

scotoglu commented on app/src/main/java/com/example/narutoquiz/data/model/Family.kt:
```diff
@@ -0,0 +1,74 @@
+import com.fasterxml.jackson.annotation.JsonProperty
+
+data class Family(
+    @JsonProperty("adoptive brother")
```
JsonProperty tüm alanlar için kullanmak zorunda mı kaldık? Sorun yaşanılan alanlar dışında kaldırsak hata veriyor mu?

yakupkavak replied:
```
 fun provideRetrofitService(mapper: ObjectMapper): NarutoService {
        return Retrofit.Builder().addConverterFactory(JacksonConverterFactory.create(mapper))
            .baseUrl(BuildConfig.BASE_URL)
            .build()
            .create(NarutoService::class.java)
    }
```
bunun içerisinde jackson tanımladığım için gson kullandığım zaman hata veriyordu

scotoglu commented on app/src/main/java/com/example/narutoquiz/data/model/Height.kt:
```diff
@@ -0,0 +1,18 @@
+    @JsonProperty("Gaiden")
+    val gaiden: String?,
+    @JsonProperty("Part I")
+    val partI: String?,
```
partOne, partTwo olarak isimlendirebiliriz.

scotoglu commented on app/src/main/java/com/example/narutoquiz/data/model/Weight.kt:
```diff
@@ -0,0 +1,10 @@
+
+data class Weight(
+    @JsonProperty("Part I")
+    val partI: String?,
```
partOne, partTwo

scotoglu commented on app/src/main/java/com/example/narutoquiz/data/util/Resource.kt:
```diff
@@ -3,7 +3,7 @@ package com.example.narutoquiz.data.util
 
     companion object {
-        fun <T> success(data: T?): Resource<T> {
+        fun <T> success(data: T): Resource<T> {
```
nullable olması daha güvenli olmaz mı?

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/extension/UtilExt.kt:
```diff
@@ -0,0 +1,6 @@
+package com.example.narutoquiz.ui.extension
+
+fun getFourRandomNumber(range: Int): List<Int>{
```
Generic hale getirebilirsin. `take` kısmında parametre olarak verelim mi? default değeri 4 olabilir. method ismi de `getRandomNumbers` ne dersin.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/MainScreen.kt:
```diff
@@ -4,18 +4,42 @@ import android.os.Bundle
+import dagger.hilt.android.AndroidEntryPoint
 
+@AndroidEntryPoint
 class MainScreen : AppCompatActivity() {
```
MainActivity daha anlamlı olabilir belki. Tabi sana kalmış.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/feedlist/generateRowList.kt:
```diff
@@ -0,0 +1,16 @@
+import com.example.narutoquiz.R
+import com.example.narutoquiz.data.model.FeedRowModel
+
+fun generateRowList() : List<FeedRowModel> {
```
Formatlama yapalım, bazı hizalama hataları var.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/feedlist/generateRowList.kt:
```diff
@@ -0,0 +1,16 @@
+
+fun generateRowList() : List<FeedRowModel> {
+    return listOf(
+        FeedRowModel("Classic",R.drawable.akatsuki,0),
```
named argument kullanılabilir. Okunurluğu artırır.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameFragment.kt:
```diff
@@ -5,21 +5,94 @@ import androidx.fragment.app.Fragment
+    private val binding get() = _binding!!
+    private val args: GameFragmentArgs by navArgs()
+    private val viewModel : GameViewModel by viewModels()
+    private var gameId: Int = 0
```
gameId, gameTopic viewModel üzerinde de tutabiliriz

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameFragment.kt:
```diff
@@ -5,21 +5,94 @@ import androidx.fragment.app.Fragment
+        observe(viewModel.firstOption){
+            with(binding){
+                ivOne.getUrl(it.imageUrl ?: "")
+                println(it.imageUrl)
```
kaldırabiliriz.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameFragment.kt:
```diff
@@ -5,21 +5,94 @@ import androidx.fragment.app.Fragment
+            with(binding){
+                ivOne.getUrl(it.imageUrl ?: "")
+                println(it.imageUrl)
+                tvOne.text = it.characterName ?: "Null"
```
Null yerine "" de olabilir.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameFragment.kt:
```diff
@@ -5,21 +5,94 @@ import androidx.fragment.app.Fragment
+    private fun setOnClick(){
+        with(binding){
+            fabClose.setOnClickListener {
+                findNavController().popBackStack()
```
bunun için navigation extension yazabiliriz. navigate gibi.

scotoglu commented on app/src/main/res/layout/fragment_game.xml:
```diff
@@ -1,11 +1,276 @@
+        android:orientation="horizontal"
+        app:layout_constraintGuide_percent="0.25" />
+
+    <androidx.constraintlayout.widget.Guideline
```
Guideline percentleri ayarlanmış peki farklı ekran boyutlarında tasarım nasıl gözüktüğünü kontrol ettik mi?

scotoglu commented on app/src/main/res/navigation/main_navigation.xml:
```diff
@@ -28,6 +28,12 @@
+        <argument
+            android:name="gameId"
+            app:argType="integer" />
+        <argument
```
boşluk bırakılabilir.

yakupkavak committed:
feat: make async function

yakupkavak committed:
feat: change selectedCard background

yakupkavak committed:
feat: implement true and false question

yakupkavak committed:
feat: add userRank

yakupkavak committed:
feat: handle loading

yakupkavak committed:
feat: add history layout

yakupkavak committed:
feat: handle glide error image

yakupkavak committed:
feat: create custom allert dialog

yakupkavak committed:
feat: add game over and challange mode

scotoglu commented on app/src/main/res/layout/activity_main_screen.xml:
```diff
@@ -2,20 +2,32 @@
+    tools:context=".ui.mainScreen.MainActivity">
 
     <androidx.fragment.app.FragmentContainerView
         android:id="@+id/fragmentContainerView2"
```
isimlemdirmeye dikkat edelim.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/MainActivity.kt:
```diff
@@ -0,0 +1,46 @@
+            WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
+        windowInsetsController.hide(WindowInsetsCompat.Type.systemBars())
+
+        val navController =
```
ayrı bir metoda alabiliriz. İnit ve nav listener içeren setupNavigation metodu olabilir.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameViewModel.kt:
```diff
@@ -1,4 +1,823 @@
+            )
+        ).data?.characters?.filter { character -> character.images?.isEmpty() == false }
+        val pageSize = charList?.size
+        return charList?.get(Random.nextInt(0, pageSize ?: 19)) ?: getNullCharacter()
```
null da dönebilir miydik? Neden getNullCharacters ihtiyaç duyduk?

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameViewModel.kt:
```diff
@@ -1,4 +1,823 @@
+    }
+
+    private suspend fun getRandomCharacter(): Character {
+        val charList = repository.getCharacterList(
```
439-444 arası birden işleminn yapıldığı ve sonucunda da list alınınan bir işlem, list dönen bie fonksiyona taşısak daha temiz olabilir.

yakupkavak replied:
getRandomCharacter() ile getFourRandomCharacter() içerisinde 4 farklı karakter alabiliyorum list dönen fonksiyonun içerisine taşımam durumunda getRandomCharacteri defalarca yazmam gerekirdi.

scotoglu replied:
1- Random integer değer ile servise gidiliyor
 2- Servisten gelen response içerisinde data alınıyor
 3- Data içerisinde characters alanına erişiliyor
 4- Erişilen alan images alanının boş olup olmamasına göre filtreleme yapılıyor. 
 
 İşlemler sonrasında bir images alanları dolu olan bir liste elde ediyorsun. Sonrasında da bu liste üzerinden random ile bir character alıyorsun. Yukarıdaki kısım asında list üzerinde filtreleme, son adım ise liste içerisinden bir character alma şeklinde iki farklı işlem gözüyle bakabiliriz. Bahsettiğim satırları farklı methoda aldığında bu iki işlem birbirinden ayrılmış olucak. 
 
 ```
 private suspend fun getCharList(): List<Character>? {
        return repository.getCharacterList(
            Random.nextInt(
                0,
                10
            )
        ).data?.characters?.filter { character -> character.images?.isEmpty() == false }
    }
```   
taşıyabiliriz dediğim kısımlar aslında burası. Bu şekilde yeni oluşturacağın methodu ihtiyaç olunan yerlerde çağırabilirsin yine. Kısaca methodu daha sade ve yapılan işlem açısından anlaşılabilir hale getirmiş oluyoruz. 🙂

scotoglu replied:
Not: Yeni methodunu yine `getRandomCharacter` içerisinde kullanıcağı için, bu metodun çağrılıcağı yerleri ve sayısını etkilemiycektir.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameViewModel.kt:
```diff
@@ -1,4 +1,823 @@
+
+    private suspend fun getRandomCharacter(): Character {
+        val charList = repository.getCharacterList(
+            Random.nextInt(
```
Randon.nextInt'ı farklı parametlerle birden fazla yerde kullanılmış, extension metoda alınabilir.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameViewModel.kt:
```diff
@@ -1,4 +1,823 @@
+
+        if (firstCharacter != null) {
+            _questionText.postValue(
+                "Which one's clan name" + " is ${firstCharacter.personal?.clan?.get(0)}"
```
String kısmını constant ya da resource ekleyip kullanılabilir.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameViewModel.kt:
```diff
@@ -1,4 +1,823 @@
+
+        trueAnswerId = options[0].optionId
+
+        if (firstCharacter != null) {
```
if'ler azaltmak için yapılan işlemler fonksiyona taşınabilir. İndex, default değerleri parametre olarak verebilirsin. Dönüş tipide Character olsun. Bu şekilde sadece uygun parametreleri verip, fonksiyonu çağırmak yeterli olur gibi geldi

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameViewModel.kt:
```diff
@@ -1,4 +1,823 @@
+                }
+            },
+            onError = {
+                println("error allert")
```
Kullanıcıya bir mesaj göstermek uygun olabilir. Toast mesajı mesela.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameViewModel.kt:
```diff
@@ -1,4 +1,823 @@
+                    }
+                }
+            },
+            onError = null,
```
default null yaparsan, null vermene gerek kalmaz.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameViewModel.kt:
```diff
@@ -1,4 +1,823 @@
+            OptionModel(2, _thirdOption),
+            OptionModel(3, _lastOption)
+        ).shuffled()
+        val firstCharacter = characterList[0]
```
0,1,2,3 değerlerini anlamlı olucak şekilde constanta verebilriz.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameViewModel.kt:
```diff
@@ -1,4 +1,823 @@
+
+        if (firstCharacter != null) {
+            _questionText.postValue(
+                "Which one's team name" + " is ${firstCharacter.personal?.team?.get(0)}"
```
constant ya da resourcedan alınabilir.

yakupkavak replied:
viewModel içerisinden nasıl getString tarzı bir yaklaşım sergilemeliyim.https://stackoverflow.com/questions/47628646/how-should-i-get-resourcesr-string-in-viewmodel-in-android-mvvm-and-databindi buraya baktığımda 
`Resource string manipulation belongs the View layer, not ViewModel layer.
`böyle bir yazı gördüm de bunu dailyde konuşabilir miyiz

yakupkavak replied:
burada dil desteği nasıl yapabiliriz not aldım konuşmak için

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameViewModel.kt:
```diff
@@ -1,4 +1,823 @@
+        }
+        trueAnswerId = options[0].optionId
+
+        if (firstCharacter != null) {
```
buradaki if'ler azaltmak için de ayrı bir fonksiyona taşıma işini yapabiliriz.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameViewModel.kt:
```diff
@@ -1,4 +1,823 @@
+    }
+
+    private suspend fun getFourTeamCharacter(): Resource<List<Character?>> {
+        val teamList = repository.getTeamList(150)
```
150, 149 sayıları okunurluğu azaltıyor. Artırmak için anlamlı olucak şekilde constant olarak tanımlayıp kullanabilriz.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameViewModel.kt:
```diff
@@ -1,4 +1,823 @@
+        val thirdTeam = teamList.data?.teams?.get(teamIdList[2])
+        val lastTeam = teamList.data?.teams?.get(teamIdList[3])
+        val firstCharacter =
+            firstTeam?.characters?.get(Random.nextInt(0, firstTeam.characters.size))
```
Random.nextInt extensiın olarak alınabilir.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameViewModel.kt:
```diff
@@ -1,4 +1,823 @@
+                    }
+                }
+            },
+            onError = null,
```
default null yapınca null vermeye gerek kalmıycak

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameViewModel.kt:
```diff
@@ -1,4 +1,823 @@
+            OptionModel(3, _lastOption)
+        ).shuffled()
+
+        val firstCharacter = characterList[0]
```
0,1,2,3 anlamlı olucak şekilde constantan alabilir. Daha önceki kullanımlarla da aynı ise genel bir isimlendirme yapılabilir.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameViewModel.kt:
```diff
@@ -1,4 +1,823 @@
+        }
+        trueAnswerId = options[0].optionId
+
+        if (firstCharacter != null) {
```
+1 if leri azaltmak için.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameViewModel.kt:
```diff
@@ -1,4 +1,823 @@
+
+    private suspend fun getFourTailCharacter(): Resource<List<Character?>> {
+        val tailList = repository.getTailedBeastList()
+        val tailIdList = getRandomNumList(4, 9)
```
namedArgument kullanmak okunurluğu artırır :)

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/feedlist/UtilFunc.kt:
```diff
@@ -0,0 +1,34 @@
+import com.example.narutoquiz.R
+import com.example.narutoquiz.data.model.Character
+import com.example.narutoquiz.data.model.FeedRowModel
+
```
File ismi Util olabilr.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/extension/UtilExt.kt:
```diff
@@ -0,0 +1,17 @@
+        listOf()
+    }
+}
+fun CardView.setBackground(context: Context,resource: Int){
```
View classına ait extension olarak kullanabiliriz. Ayrıca args olarak context vermeyebilirsin, view contextte kullanılabilir. (Parametreyi kaldırınca fark edersin :)

ViewExt ayrı bir file da taşıyabiliriz.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/base/BaseViewModel.kt:
```diff
@@ -8,24 +8,30 @@ import kotlinx.coroutines.launch
+                Status.LOADING -> {
+                    onLoading?.invoke()
+                }
             }
```
Formatlama yapalım. Ctrl + Alt + L

scotoglu commented on app/src/main/java/com/example/narutoquiz/data/di/NetworkModule.kt:
```diff
@@ -22,6 +39,4 @@ object NetworkModule {
     fun provideRepository(pokeService: NarutoService): NarutoRepository {
```
pokeService?? :)

scotoglu commented on app/src/main/java/com/example/narutoquiz/domain/extension/UtilExt.kt:
```diff
@@ -1,27 +1,56 @@
+import com.example.narutoquiz.data.model.VoiceActors
+import kotlin.reflect.KProperty1
 
 fun ImageView.getUrl(url: String) {
```
GlideExt file taşıyalım

scotoglu commented on app/src/main/res/layout/activity_sign.xml:
```diff
@@ -2,7 +2,8 @@
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
-    android:id="@+id/main"
+    android:background="@drawable/gradient_background_color"
```
Ekrandaki warninglere dikkat edelim.

scotoglu commented on app/src/main/res/layout/fragment_signin.xml:
```diff
@@ -1,7 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Daha önce de konuşmuştuk, tekrar not olması için yazıyorum. ScrollView alınması iyi olabilr.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/userLogIn/login/SignUpFragment.kt:
```diff
@@ -64,7 +64,7 @@ class SignUpFragment : Fragment() {
                                 Toast.LENGTH_LONG
                             ).show()
-                            Intent(requireContext(),MainScreen::class.java).also {
+                            Intent(requireContext(),MainActivity::class.java).also {
```
Neden activity olarak başlatıyoruz?

yakupkavak committed:
refactor: add const

yakupkavak committed:
refactor: add const

yakupkavak committed:
refactor: add function

yakupkavak committed:
refactor: complete view model functions

yakupkavak committed:
bugfix: solve setOption index problem

yakupkavak committed:
refactor: design code

yakupkavak committed:
feat: add Error Dialog

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/main/ErrorDialogFragment.kt:
```diff
@@ -0,0 +1,30 @@
+) : DialogFragment() {
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val binding: DialogGameBinding = DataBindingUtil.inflate(
+            this.requireActivity().layoutInflater, R.layout.dialog_error, null, false
```
java.lang.ClassCastException: com.example.narutoquiz.databinding.DialogErrorBindingImpl cannot be cast to com.example.narutoquiz.databinding.DialogGameBinding

yakupkavak committed:
feat: add FirebaseRepository

yakupkavak committed:
fix: solve select true answer problem

Added setOptionTrue and setOptionWrong

yakupkavak committed:
feat: add Firerepository and complete sign functions

yakupkavak committed:
feat: add history list

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/extension/Util.kt:
```diff
@@ -0,0 +1,48 @@
+fun getRandom(from: Int = 0, includeUntil: Int): Int{
+    return Random.nextInt(from,includeUntil+1)
+}
+fun Context.setString(gameId: Int) : String{
```
Ayrı bir file alalım, GameUtils vs olabilir. Extension ismide setStringByGameId olabilir. Çünkü setSetring genel bir kullanım gibi geliyor ama id bağımlığı olan, belirli bir case için tanımlanmış.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/extension/Util.kt:
```diff
@@ -0,0 +1,48 @@
+            this.getString(R.string.team)
+        }
+        else -> {
+            return "Null"
```
return eklemene gerek yok, when zaten return ediyor. "Null" yerine de null yada Empty("") dönelim.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameFragment.kt:
```diff
@@ -5,21 +5,304 @@ import androidx.fragment.app.Fragment
+    }
+
+    private fun setObserve() {
+        viewModel.startGame()
```
İhtiyacına göre viewModel.init yada onViewCreated çağırabilirsin. setObserve içinde çağırmayalım. Benzer şekilde diğer yerlerde de kullanılmış, onları da kontrol edelim.

yakupkavak replied:
History ve Rank sayfasında init olarak yaptığım zaman ilk başta doğru gösteriyor ama kullanıcı oyun oynadıktan sonra tekrar ekranı açınca eski bilgiler gözükmüyor. her açıldığında güncel olması için init kullanmadım

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameFragment.kt:
```diff
@@ -5,21 +5,304 @@ import androidx.fragment.app.Fragment
+        observe(viewModel.error) {
+            if (it) {
+                val newFragment = ErrorDialogFragment(
+                    mainScreen = { navigate(GameFragmentDirections.actionGameFragmentToFeedFragment()) })
```
mainScreen yerine onClick vs verebilirsin. Burada anladığım kadarıyla bir click aksiyonu söz konusu bu şekilde belirtmek daha anlamlı olucaktır.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameFragment.kt:
```diff
@@ -5,21 +5,304 @@ import androidx.fragment.app.Fragment
+        observe(viewModel.answerSelection) {
+            it.trueAnswer?.let { trueAnswerId ->
+                when (trueAnswerId) {
+                    0 -> {
```
Constantlara ekleyelim.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameFragment.kt:
```diff
@@ -5,21 +5,304 @@ import androidx.fragment.app.Fragment
+                cvFour.setOnClickListener {
+                    clearSelection()
+                    cvFour.setBackground(R.color.selected_answer)
+                    selectedOptionId = 3
```
3 neyi ifade ediyor. Anlamlı olması için constanttan alalım.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameFragment.kt:
```diff
@@ -5,21 +5,304 @@ import androidx.fragment.app.Fragment
+            }
+            it.falseAnswer?.let { falseAnswerId ->
+                when (falseAnswerId) {
+                    0 -> {
```
Constanttan alalım

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameFragment.kt:
```diff
@@ -5,21 +5,304 @@ import androidx.fragment.app.Fragment
 
+    private fun setInvisible() {
+        with(binding) {
+            tvQuestion.isVisible = false
```
ConstatainLayout -> Group kullanımı işini kolalaştıracaktır. Tek satırı düşürebilirsin group ile.

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/game/GameFragment.kt:
```diff
@@ -5,21 +5,304 @@ import androidx.fragment.app.Fragment
+
+    private fun setVisible() {
+        with(binding) {
+            tvQuestion.isVisible = true
```
Group ile aynı şekilde. +1

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/history/HistoryFragment.kt:
```diff
@@ -0,0 +1,55 @@
+        }
+        observe(viewModel.loading){
+            if (it){
+                println("yükleniyor")
```
Ekranda progress ya da mesaj göstermek daha doğru olur. Not olarak kalsın

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/mainScreen/rankList/RankFragment.kt:
```diff
@@ -0,0 +1,57 @@
+        }
+        observe(viewModel.loading){
+            if (it){
+                println("yükleniyor")
```
+1

scotoglu commented on app/src/main/java/com/example/narutoquiz/ui/userLogIn/recovery/RecoveryFragment.kt:
```diff
@@ -25,12 +32,36 @@ class RecoveryFragment : Fragment() {
+    private fun setObserve() {
+        observe(viewModel.resetSuccess) {
+            if (it) {
+                Toast.makeText(requireContext(), getString(R.string.check_mail), Toast.LENGTH_LONG)
```
FragmentExt içine extension olarak yazabilirsin. showToast isimli method olabilir.

scotoglu commented on app/src/main/res/layout/fragment_game.xml:
```diff
@@ -1,11 +1,303 @@
 
+    <com.airbnb.lottie.LottieAnimationView
+        android:id="@+id/lottie_animation_naruto"
+        android:layout_width="128dp"
```
dimens.xml'den alalım.

yakupkavak committed:
fix: solve historyList

yakupkavak committed:
refactor: try scroolview

yakupkavak committed:
feat: show only highest gameScore

yakupkavak committed:
feat: get hint and refactor code

yakupkavak committed:
feat: add info dialog

yakupkavak committed:
refactor: change toast

yakupkavak committed:
feat: add market page layout

yakupkavak committed:
feat: create SharedViewModel

yakupkavak committed:
feat: add GameRoomDatabase

yakupkavak committed:
feat: get history with Room Database

yakupkavak committed:
bugfix: solve history problem.

When we change user, it still shows past user history. It solved.

yakupkavak committed:
feat: add rankList firestore

Now in the rankList only the highest score of the user point can be save.

yakupkavak committed:
bugfix: solve show hint count problem

Change cloud function.

yakupkavak committed:
feat: implemet tokenCount in GameFragment

yakupkavak committed:
bugfix: solve sharedViewModel

define activityViewModels in Game and UserInfo fragments

yakupkavak committed:
feat: change password and e mail text

yakupkavak committed:
feat: add firebase app check

yakupkavak committed:
feat: initalize adMob

yakupkavak committed:
feat: add remote config

yakupkavak committed:
feat: implement remote config with about game

yakupkavak committed:
feat: buy product with remote config

yakupkavak committed:
feat: create dialog for successful sale

yakupkavak committed:
feat: add product price with remote config

yakupkavak committed:
feat: secure api key

yakupkavak committed:
feat: change password completed

To change password I use reauth

yakupkavak committed:
refactor: show product
with searching product name not id.

yakupkavak committed:
feat: buy product with consume purchase

yakupkavak committed:
refactor: change project name

yakupkavak committed:
feat: complete in app purchase.

yakupkavak committed:
feat: delete account

yakupkavak committed:
feat: complete add hint with watch ad

yakupkavak committed:
refactor: change gemini api and admob api

yakupkavak committed:
feat: change challange game layout

yakupkavak committed:
feat: can stay sign in

yakupkavak committed:
bugfix: solve buy product problem

The mistake is different parameter name.

yakupkavak committed:
feat: add icon

yakupkavak committed:
refactor: delete unnecessary package names.

yakupkavak committed:
feat: update question types

yakupkavak committed:
bugfix: solve hint buy problem

There is an initialize time problem. solve with add it to setData. When update it on marketViewModel also update it SharedViewModel.

yakupkavak committed:
feat: add icon to first three player and change their background color

yakupkavak committed:
feat: change game score style with user's current point

yakupkavak committed:
refactor: change constant names

yakupkavak committed:
feat: add loading station while ad loading.

yakupkavak committed:
refactor: delete ask voice actor.

It is so hard to know by users.

yakupkavak committed:
feat: add loading animation on sign in.

yakupkavak committed:
bugfix: solve family and leader problem.

The family range was wrong, didn't check is it challenge.

yakupkavak committed:
bugfix: solve rank problem. IDE can't clear past binds.

Add clear function for all of them.

yakupkavak committed:
feat: create extension to load gif
